lib/src/main/java/com/shopify/checkoutsheetkit/CheckoutWebView.kt

```diff
@@ -239,24 +238,49 @@ internal class CheckoutWebView(context: Context, attributeSet: AttributeSet? = n
                 handleError(
                     request,
                     errorResponse.statusCode,
-                    errorResponse.reasonPhrase
+                    errorResponse.reasonPhrase,
+                    errorResponse.responseHeaders,
                 )
             }
         }
 
+        private fun isRecoverable(statusCode: Int): Boolean {
+            return when (statusCode) {
+                TOO_MANY_REQUESTS -> true
+                in CLIENT_ERROR -> false
+                else -> true
+            }
+        }
+
         private fun handleError(
             request: WebResourceRequest?,
             errorCode: Int,
-            errorDescription: String
+            errorDescription: String,
+            responseHeaders: MutableMap<String, String> = mutableMapOf(),
         ) {
             if (request?.isForMainFrame == true) {
-                val exception = when (errorCode) {
-                    HTTP_NOT_FOUND -> CheckoutLiquidNotMigratedException()
-                    HTTP_GONE -> CheckoutExpiredException()
-                    HTTP_INTERNAL_ERROR -> CheckoutUnavailableException()
-                    else -> CheckoutSdkError(errorDescription)
+                val processor = checkoutBridge.getEventProcessor()
+                when {
+                    errorCode == HTTP_NOT_FOUND && responseHeaders[DEPRECATED_REASON_HEADER] == LIQUID_NOT_SUPPORTED -> {
+                        processor.onCheckoutViewFailedWithError(
+                            ConfigurationException(
+                                errorDescription = "The checkout URL provided has resulted in an error. The store is still using " +
+                                "checkout.liquid, whereas the checkout SDK only supports checkout with extensibility.",
```

Comment:
I've changed this message to this in the swift lib:

```suggestion
                                errorDescription = "Storefronts using checkout.liquid are not supported. Please upgrade to Checkout Extensibility.",
```